File: 프로그래머스/lv1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.js
function solution(n) {
    let min = Math.min(...n)  // 1      얘의 위치를 찾아야 함. filter  !min === ele
    if (n.length > 1) {
        let idx = n.indexOf(min) // 원본을 바꾸는 메서드
        n.splice(idx,1)
        return n
    } else {
        return [-1]
    }
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
class Solution {
    public int[] solution(int[] arr) {
        // 원소가 하나라면 [-1] 리턴
        if (arr.length == 1) {
            return new int[]{-1};
        }

        // 제일 작은 수 찾기
        int min = arr[0];
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] < min) {
                min = arr[i];
            }
        }

        // 가장 작은 수를 뺀 새 배열 만들기
        int[] answer = new int[arr.length - 1];
        int idx = 0;
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] != min) {
                answer[idx] = arr[i];
                idx++;
            }
        }

        return answer;
    }
}
